bindgen/ir/context.rs

```diff
@@ -2081,55 +2083,79 @@ If you encounter an error missing from this list, please file an issue or a PR!"
 
             let index = clang::Index::new(false, false);
 
-            let mut c_args = Vec::new();
-            let mut pch_paths = Vec::new();
-            for input_header in self.options().input_headers.iter() {
-                let path = Path::new(input_header.as_ref());
-                let header_name = path
-                    .file_name()
-                    .and_then(|hn| hn.to_str())
-                    .map(|s| s.to_owned());
-                let header_path = path
-                    .parent()
-                    .and_then(|hp| hp.to_str())
-                    .map(|s| s.to_owned());
-
-                let (header, pch) = if let (Some(ref hp), Some(hn)) =
-                    (header_path, header_name)
-                {
-                    let header_path = if hp.is_empty() { "." } else { hp };
-                    let header = format!("{header_path}/{hn}");
-                    let pch_path = if let Some(ref path) =
-                        self.options().clang_macro_fallback_build_dir
-                    {
-                        path.as_os_str().to_str()?
-                    } else {
-                        header_path
-                    };
-                    (header, format!("{pch_path}/{hn}.pch"))
-                } else {
-                    return None;
-                };
-
-                let mut tu = clang::TranslationUnit::parse(
-                    &index,
-                    &header,
-                    &[
-                        "-x".to_owned().into_boxed_str(),
-                        "c-header".to_owned().into_boxed_str(),
-                    ],
-                    &[],
-                    clang_sys::CXTranslationUnit_ForSerialization,
+            let (header_names, header_paths, header_contents) =
+                self.options.input_headers.iter().try_fold(
```

Comment:
Is there a good reason to use try_fold here rather than just a for loop? Maybe I'm old-fashioned, but I find a for loop easier to reason about :)

Reply:
Resolved.